src/init.js

```diff
@@ -41,6 +42,8 @@ function init(options) {
     var fileContent = file.contents.toString();
     var sourceMap;
 
+    file.newline = detectNewline.graceful(fileContent || '');
```

Comment:
Good idea,  I am working on some other workarounds by tackingon fields on file as well. However,  we may want to stick to modifying the sourceMap field/object to ha e less possible collisions.